clients/src/main/java/org/apache/kafka/common/metrics/JmxReporter.java

```diff
@@ -133,7 +133,7 @@ static String getMBeanName(String prefix, MetricName metricName) {
             mBeanName.append(",");
             mBeanName.append(entry.getKey());
             mBeanName.append("=");
-            mBeanName.append(entry.getValue());
+            mBeanName.append(ObjectName.quote(entry.getValue()));
```

Comment:
This is an externally visible change since client-id appears quoted in metrics and can break users who rely on the existing un-quoted value.